File: 210818/C++/BOJ_4659.cpp
#include<bits/stdc++.h>
using namespace std;

int N;

// 모음 포함 O
// 연속글자X But ee, oo 허용
// 3연속 자음 or 모음 X
bool chkMo(char ch){
    if(ch=='a' || ch=='e' || ch=='i' || ch=='o' || ch=='u') return true;
    return false;
}

int main() {
    ios_base::sync_with_stdio(0);
    cin.tie(0);
    string str;
    while(1){
        cin>>str;
        if(str=="end") break;

        bool res = true;
        bool mo = false;
        for(int i=0; i<str.size(); i++){
            if(chkMo(str[i])) mo = true;
            if(i<str.size()-1){
                if(str[i]==str[i+1]) {
                    if(str[i]=='e' || str[i]=='o');
                    else {
                        res = false;
                        break;
                    }
                }
            }
            if(i<str.size()-2){
                bool x = chkMo(str[i]);
                bool y = chkMo(str[i+1]);
                bool z = chkMo(str[i+2]);
                if((x && y && z) || (!x && !y && !z)){
                    res = false;
                    break;
                }
            }
        }
        
        if(res && mo)
            cout<<'<'<<str<<'>'<<" is acceptable."<<'\n';
        else
            cout<<'<'<<str<<'>'<<" is not acceptable."<<'\n';
    }
    
}

File: 210818/Java/BOJ_4659.java
import java.io.BufferedReader;
import java.io.InputStreamReader;
import java.util.StringTokenizer;

public class Main {

    public static void main(String[] args) throws Exception{
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
        StringTokenizer st;

        String str;
        while(true){
            st = new StringTokenizer(br.readLine());
            str = st.nextToken();
            if(str.equals("end")) break;

            boolean res = true;
            boolean mo = false;
            for(int i=0; i<str.length(); i++){
                if(chkMo(str.charAt(i))) mo = true;
                if(i<str.length()-1){
                    if(str.charAt(i)==str.charAt(i+1)) {
                        if(str.charAt(i)=='e' || str.charAt(i)=='o');
                        else {
                            res = false;
                            break;
                        }
                    }
                }
                if(i<str.length()-2){
                    boolean x = chkMo(str.charAt(i));
                    boolean y = chkMo(str.charAt(i+1));
                    boolean z = chkMo(str.charAt(i+2));
                    if((x && y && z) || (!x && !y && !z)){
                        res = false;
                        break;
                    }
                }
            }

            if(res==true && mo) System.out.println('<' + str + '>' + " is acceptable.");
            else System.out.println('<' + str + '>' + " is not acceptable.");
        }

    }

    static boolean chkMo(char ch){
        if(ch=='a' || ch=='e' || ch=='i' || ch=='o' || ch=='u') return true;
        return false;
    }
}